feat: Add filtering and pagination functionality

tamerlantian opened the pull request:
- Implemented new feature to allow filtering and pagination on mongodb data-source
- Enhance Swagger documentation for pagination and state filtering on user endpoint.

tamerlantian committed:
feat: Add filtering and pagination functionality

-Implemented new feature to allow filtering and pagination on mongodb data-source
- Updated Swagger documentation to support pagination and state filtering on user endpoint

thEpisode commented on package.json:
```diff
@@ -2,7 +2,7 @@
   "title": "Loom service",
   "description": "Another Node.js Server framework to create microservices or huge monoliths.",
-  "version": "6.1.10",
+  "version": "6.2.10",
```
version is 6.1.11

thEpisode commented on src/data-sources/mongodb/mongodb-data-source.js:
```diff
@@ -113,32 +146,90 @@ class MongoDBDataSource extends DataSource {
+   */
+  #extractPagination(filters) {
+    const pagination = {
+      skip: null,
```
add a default value

tamerlantian replied:
Null is required, as in case the URL does not include any 'skip' or 'limit' fields, a validation will be triggered, and it will return all the data in the collection.

thEpisode commented on src/services/user/user-management/user-management.service.js:
```diff
@@ -125,6 +128,39 @@ class UserService {
+      const result = await this.#getByFilters({
+        filters: [
+          { key: 'status.name', operator: 'in', value: data.include_status },
+          {
```
use the same code developing convention, key, operator and value in the same line

tamerlantian replied:
![image](https://github.com/link-loom/loom-svc-js/assets/70783227/691cd2a8-1ffe-48b2-9936-ca8a555356d7)

ESLint flags this as an error when I adhere to the convention you mentioned earlier

thEpisode commented on src/data-sources/mongodb/mongodb-data-source.js:
```diff
@@ -96,13 +104,38 @@ class MongoDBDataSource extends DataSource {
+      }
+
+      if (pagination.limit !== null) {
+        dataPipeline.push({ $limit: +pagination.limit });
```
this is not a safety conversion, eg, +'h' = NaN

thEpisode commented on src/data-sources/mongodb/mongodb-data-source.js:
```diff
@@ -96,13 +104,38 @@ class MongoDBDataSource extends DataSource {
+          {
+            $project: {
+              data: '$data',
+              matchCount: { $arrayElemAt: ['$matchCount.total', 0] },
```
Set default values but not fixed values

tamerlantian committed:
Address review comments for PR #30

- Bump project version in package.json.
- Remove unsafe operator '+' from skip and limit conversion variables.
- Remove fixed values in getByFilters and set default values.
- Documentation improved

tamerlantian committed:
Address review comments for PR #30

- Renamed variables for better clarity and descriptiveness.
- Replaced the '+' operator with `Number()` for type conversion in the `extractPagination` function logic.
